app/src/main/java/org/mozilla/focus/search/MultiselectSearchEngineListPreference.java

```diff
@@ -48,4 +49,36 @@ protected void updateDefaultItem(CompoundButton defaultButton) {
         }
         return engineIdSet;
     }
+
+    // Whenever an engine is checked or unchecked, we adapt the MenuItem state (enable or disable) given in param
+    // Though some conditions are added just for optimisation
+    public void bindEnginesCheckboxesTo(final MenuItem menuDeleteItem, final int alpha_default, final int alpha_greyed) {
```

Comment:
\[PMD\] INFO: The method 'bindEnginesCheckboxesTo' has a Cyclomatic Complexity of 7\.